src/clients/api/queries/useGetPrimeEstimation/index.ts

```diff
@@ -0,0 +1,212 @@
+    symbol: 'XVS',
+  });
+
+  const noEmptyValues = !!borrowedAmountTokens && !!stakedAmountXvsTokens && !!suppliedAmountTokens;
+  const enabled = !!options?.enabled && noEmptyValues && !!vToken;
+
+  const { data: primeDistributionForMarketData } = useGetPrimeDistributionForMarket(
+    {
+      vTokenAddress: vToken?.address || '',
+    },
+    {
+      enabled: !!vToken,
+    },
+  );
+
+  const primeTokensDistributedAmount = convertMantissaToTokens({
+    value: primeDistributionForMarketData?.totalDistributedMantissa || new BigNumber(0),
+    token: vToken?.underlyingToken,
+  });
+
+  const { userSupplyBalanceMantissa, userBorrowBalanceMantissa, userXvsStakedMantissa } =
+    useMemo(() => {
+      const userSupplyBalance = vToken
+        ? convertTokensToMantissa({
+            value: new BigNumber(suppliedAmountTokens),
+            token: vToken.underlyingToken,
+          })
+        : new BigNumber(0);
+      const userBorrowBalance = vToken
+        ? convertTokensToMantissa({
+            value: new BigNumber(borrowedAmountTokens),
+            token: vToken.underlyingToken,
+          })
+        : new BigNumber(0);
+      const userXvsStaked = xvs
+        ? convertTokensToMantissa({
+            value: new BigNumber(stakedAmountXvsTokens),
+            token: xvs,
+          })
+        : BigNumber(0);
+
+      return {
+        userSupplyBalanceMantissa: userSupplyBalance,
+        userBorrowBalanceMantissa: userBorrowBalance,
+        userXvsStakedMantissa: userXvsStaked,
+      };
+    }, [suppliedAmountTokens, borrowedAmountTokens, stakedAmountXvsTokens, vToken, xvs]);
+
+  const { data: hypotheticalPrimeApysData } = useGetHypotheticalPrimeApys(
+    {
+      vTokenAddress: vToken?.address || '',
+      userSupplyBalanceMantissa,
+      userBorrowBalanceMantissa,
+      userXvsStakedMantissa,
+      accountAddress: accountAddress || NULL_ADDRESS,
+    },
+    {
+      enabled,
+    },
+  );
+
+  const estimation = useMemo(() => {
+    let primeEstimation: UseGetPrimeEstimationOutput = {
+      tokensDistributedAmount: undefined,
+      borrowedTokens: undefined,
+      borrowApyPercentage: undefined,
+      borrowCapTokens: undefined,
+      borrowCapCents: undefined,
+      suppliedTokens: undefined,
+      supplyApyPercentage: undefined,
+      supplyCapTokens: undefined,
+      supplyCapCents: undefined,
+      userYearlyPrimeRewards: undefined,
+    };
+
+    if (hypotheticalPrimeApysData && vToken && xvs) {
+      const {
+        borrowApyPercentage,
+        borrowCapMantissa,
+        borrowCapUsd: borrowCapPriceMantissa,
+        supplyApyPercentage,
+        supplyCapMantissa,
+        supplyCapUsd: supplyCapPriceMantissa,
+        userPrimeRewardsShare,
+      } = hypotheticalPrimeApysData;
+
+      const borrowedTokens = new BigNumber(borrowedAmountTokens);
+
+      const suppliedTokens = new BigNumber(suppliedAmountTokens);
+
+      const supplyCapUsd = convertPriceMantissaToDollars({
```

Comment:
I don't think there should be functions that convert from "calculation" units (units that are used for calculation but not displayed) to doing a calculation and converting to "display" units.
Calculation functions should return "calculation units" and then we can convert calculation to display units. In this way you remove the need for the function on line 161 and have clear separation of responsibilities

Reply:
In this case the value returned is in USD so the order should be clean incoming values to cents in `useGetHypotheticalPrimeApys` first. And then consumers of that query have the values in the correct unit to do calculations with until it is time to display a value